src/lib/Microsoft.Health.Events/Common/EventHubFormatter.cs

```diff
@@ -0,0 +1,33 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+
+namespace Microsoft.Health.Events.Common
+{
+    public static class EventHubFormatter
+    {
+        public static string GetEventHubFQDN(string host)
+        {
+            EnsureArg.IsNotEmptyOrWhiteSpace(host);
+
+            if (Uri.IsWellFormedUriString(host, UriKind.Absolute))
+            {
+                var uri = new Uri(host);
+                host = uri.Host;
+            }
+
+            if (Uri.CheckHostName(host) != UriHostNameType.Unknown)
```

Comment:
Nit, I personal prefer put validations first and let it fail as early as possible.

e.g.

if (!Uri.IsWellFormedUriString(host, UriKind.Absolute)) {
   throw new Exception("not well formatted.");
}

var uri = new Uri(host);
host = uri.Host;

if(Uri.CheckHostName(host) == UriHostNameType.Unknown)
 throw new Exception("host is unknown");
)

return host;